modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/JoinCommutePlannerTest.java

```diff
@@ -76,13 +79,44 @@ public static void init() {
                                 .add("ID", TYPE_FACTORY.createJavaType(Integer.class))
                                 .build(), 10) {
 
-                    @Override public IgniteDistribution distribution() {
+                    @Override
+                    public IgniteDistribution distribution() {
                         return IgniteDistributions.affinity(0, UUID.randomUUID(), DEFAULT_ZONE_ID);
                     }
                 }
         );
     }
 
+    @Test
+    public void testEnforceJoinOrderHint() throws Exception {
+        String sqlJoinCommuteWithNoHint = "SELECT COUNT(*) FROM SMALL s, HUGE h, HUGE h1 WHERE h.id = s.id and h1.id=s.id";
+        String sqlJoinCommuteWithHint =
+                "SELECT /*+ ENFORCE_JOIN_ORDER */ COUNT(*) FROM SMALL s, HUGE h, HUGE h1 WHERE h.id = s.id and h1.id=s.id";
+        String sqlJoinCommuteOuterWithNoHint = "SELECT COUNT(*) FROM SMALL s RIGHT JOIN HUGE h on h.id = s.id";
+        String sqlJoinCommuteOuterWithHint = "SELECT /*+ ENFORCE_JOIN_ORDER */ COUNT(*) FROM SMALL s RIGHT JOIN HUGE h on h.id = s.id";
+
+        RuleAttemptListener listener = new RuleAttemptListener();
+
+        physicalPlan(sqlJoinCommuteWithNoHint, publicSchema, listener);
```

Comment:
i suggest to use real rule names instead of hadcoded, one change for rule namings will broke huge amount of tests :
JoinConditionPushRuleConfig rule = JoinConditionPushRuleConfig.DEFAULT;
String ruleName = rule.toRule().toString();

Reply:
rewrote, thanks